settings.gradle.kts

```diff
@@ -351,6 +352,35 @@ gradle.rootProject {
         outputFile = layout.projectDirectory.file("architecture/platforms.md")
         elements = provider { architectureElements.map { it.build() } }
     }
+    tasks.register("platformsData", GeneratePlatformsDataTask::class) {
+        description = "Generates the platforms data"
+        outputFile = layout.projectDirectory.file("build/architecture/platforms.json")
+        platforms = provider { architectureElements.filterIsInstance<PlatformBuilder>().map { it.build() } }
+    }
+}
+
+abstract class GeneratePlatformsDataTask : DefaultTask() {
+
+    data class PlatformData(val name: String, val dirs: List<String>, val uses: List<String>)
+
+    @get:OutputFile
+    abstract val outputFile: RegularFileProperty
+
+    @get:Input
+    abstract val platforms: ListProperty<Platform>
+
+    @TaskAction
+    fun action() {
+        val allPlatforms = platforms.get()
+        val data = allPlatforms.map { platform ->
```

Comment:
🤔 It seems like we should define this platform structure data outside of any particular task, and use it as an `@Input` to this task.  Otherwise, we won't know when to skip updating the outputFile when the structure is up to date.